units/dim.cpp

```diff
@@ -3,49 +3,36 @@
 /// @file
 /// @author Simon Heybrock
 #include <limits>
+#include <mutex>
 
 #include "scipp/common/index.h"
 #include "scipp/units/dim.h"
 
 namespace scipp::units {
 
 std::unordered_map<std::string, Dim::Id> Dim::builtin_ids{
```

Comment:
Does it make sense to keep the two separate maps? How expensive is the lookup in `builtin_ids` compared to acquiring the shared lock?

Reply:
I don't know. I suppose we can figure that out separately?